### DISCUSSION TITLE
Connecting to two databases

### DISCUSSION THREAD

**Discussion by BJladika** [prisma/prisma#7565] [Q&A] [answered]
So, if I'm now following https://github.com/prisma/prisma-examples/tree/latest/javascript/graphql-sdl-first
and If I need to connect to two databases (like sqlite and mssql server) I need just basically follow an example from this discussion?
https://github.com/prisma/prisma/issues/2443#issuecomment-630679118

Or there are already a better ways/approaches?

(For now found https://github.com/Errorname/prisma-multi-tenant, but it seems like dead, tried to install/follow steps only to receive errors like `The package @prisma/cli has been renamed to prisma.`)

**Comment by ryands17** [comment, marked as answer]
@BJladika currently the workaround specified in the issue is the only way to connect. Separate clients and schemas for separate databases.

**Reply by BJladika** [reply]
I'm just confused with
```javascript
import { PrismaClient as PrismaClient1 } from '../prisma/client1'
import { PrismaClient as PrismaClient2 } from '../prisma/client2'
```
while on the example its 
```javascript
const { PrismaClient } = require('@prisma/client')
```
So theoretically I should exactly copy example code or replace it with smth like...
```javascript
const { PrismaClient as PrismaClient1 } = require('@prisma/client1')
const { PrismaClient as PrismaClient2 } = require('@prisma/client2')
```
?

**Reply by ryands17** [reply]
For two separate databases you will need two `schema.prisma` in different locations so in turn you have two different Prisma Clients.

**Reply by BJladika** [reply]
Ah, I got it, by my (retarded) way I must first generate prisma client(1), cut it from `node_modules/prisma/client` (or by more clever way write option of output at schema1 like) and then repeat same with second database aka client2
```prisma
generator client {
  provider = "prisma-client-js"
  output   = "./client1"
}
```

**Reply by BJladika** [reply]
Though now I tried to `npm run dev` with this all generated and now I'm encountering an error
```
import { PrismaClient as PrismaClient1 } from '../prisma/client1'
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at wrapSafe (node:internal/modules/cjs/loader:1024:16)
    at Module._compile (node:internal/modules/cjs/loader:1072:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1137:10)
    at Module.load (node:internal/modules/cjs/loader:973:32)
    at Function.Module._load (node:internal/modules/cjs/loader:813:14)
    at Module.require (node:internal/modules/cjs/loader:997:19)
    at require (node:internal/modules/cjs/helpers:92:18)
    at Object.<anonymous> (C:\Projects\graphql-sdl-first\src\server.js:3:21)
    at Module._compile (node:internal/modules/cjs/loader:1108:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1137:10)
npm ERR! code 1
npm ERR! path C:\Projects\graphql-sdl-first
npm ERR! command failed
npm ERR! command C:\Windows\system32\cmd.exe /d /s /c node ./src/server.js
```
for now my context.js looks like this
```javascript
//const { PrismaClient } = require('@prisma/client')
import { PrismaClient as PrismaClient1 } from '../prisma/client1'
import { PrismaClient as PrismaClient2 } from '../prisma/client2'

const prisma1 = new PrismaClient1({
  log: ['query', 'info', `warn`, `error`],
})

prisma1.$on('error', e => {
  console.log("Query: " + e.error)
})

const prisma2 = new PrismaClient2({log: ['query', 'info', `warn`, `error`],
})

prisma2.$on('error', e => {
  console.log("Query: " + e.error)
})

const context = {
  prisma1: prisma1,
  prisma2: prisma2
}

module.exports = {
  context,
}
```

**Reply by BJladika** [reply]
Ah, nevermind, minor googling I have replaced example code from discussion
```javascript
import { PrismaClient as PrismaClient1 } from '../prisma/client1'
import { PrismaClient as PrismaClient2 } from '../prisma/client2'
```
with
```javascript
const { PrismaClient : PrismaClient1 } = require('../prisma/client1')
const { PrismaClient : PrismaClient2 } = require('../prisma/client2')
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
